### PR TITLE
#562: output: add output to JSON for vt-tv for derived stats

### LINKED ISSUE #558: Add node-level memory constraint

### LINKED ISSUE #562: Add derived stats to output for vt-tv

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #562

Linked to:
- https://github.com/DARMA-tasking/vt-tv/pull/129
- https://github.com/DARMA-tasking/vt/pull/2376

### CODE REVIEW

File: src/lbaf/IO/lbsVTDataWriter.py [hunk 1]
```diff
@@ -134,6 +135,20 @@ def __create_tasks(self, rank_id, objects, migratable):
             user_defined = o.get_user_defined()
             if user_defined:
                 task_data["user_defined"] = dict(sorted(user_defined.items()))
+            else:
+                task_data["user_defined"] = dict()
+
+            task_data["user_defined"]["object_memory"] = o.get_size()
```
Comment (lifflander) [inline src/lbaf/IO/lbsVTDataWriter.py:141]: Can you do the same for the object-level QOIS? @cwschilly

**Comment by ppebay** [comment]
@cwschilly the merger of PR #570 has resulted in conflicts in this PR, can you please also look into those? Thanks!

File: src/lbaf/IO/lbsVTDataWriter.py (hunk 1 above)
Comment (cwschilly) [reply src/lbaf/IO/lbsVTDataWriter.py]: Done

File: src/lbaf/Model/lbsBlock.py [hunk 2]
```diff
@@ -54,10 +54,10 @@ def __init__(
             o_ids = set()
 
         # Block index
-        if not isinstance(b_id, int) or isinstance(b_id, bool):
-            raise TypeError(
-                f"b_id: incorrect type {type(b_id)}")
-        self.__index = b_id
+        # if not isinstance(b_id, int) or isinstance(b_id, bool):
```
Comment (ppebay) [inline src/lbaf/Model/lbsBlock.py:57]: Why is this commented out? Either uncomment or remove.

File: src/lbaf/Model/lbsObject.py [hunk 3]
```diff
@@ -44,7 +44,7 @@
 
 from .lbsBlock import Block
 from .lbsObjectCommunicator import ObjectCommunicator
-from .lbsQOIDecorator import qoi
+from .lbsQOIDecorator import qoi, property
```
Comment (ppebay) [inline src/lbaf/Model/lbsObject.py:47]: This causes a warning on 3.8

File: src/lbaf/Model/lbsObject.py [hunk 4]
```diff
@@ -226,17 +226,23 @@ def get_received(self) -> dict:
     @qoi
     def get_received_volume(self) -> float:
         """Return volume of communications received by object."""
-        return sum(v for v in self.__communicator.get_received().values()) if self.__communicator else 0
+        v : float = 0.0
```
Comment (ppebay) [inline src/lbaf/Model/lbsObject.py:229]: Is it necessary to hint that this is a float, as it is initialized at `0.0` immediately?

File: src/lbaf/Model/lbsObject.py [hunk 5]
```diff
@@ -289,11 +295,29 @@ def get_unused_params(self) -> dict:
...
+            name = "overhead_memory"
+        return name
+
+    def get_entity_properties(self) -> list:
+        """Get all methods decorated with the 'property' decorator.
+        """
+        property_methods : dict = {
+            self.__get_property_or_qoi_name(name): getattr(self, name)
+            for name in dir(self)
+            if callable(getattr(self, name)) and not name.startswith("__") and hasattr(getattr(self, name), "is_property") }
```
Comment (ppebay) [inline src/lbaf/Model/lbsObject.py:313]: Please add comment here to explain the logic of this test for future reference

**Review by ppebay — CHANGES_REQUESTED** [review]
Cf. comments

File: src/lbaf/Model/lbsBlock.py (hunk 2 above)
Comment (cwschilly) [reply src/lbaf/Model/lbsBlock.py]: Fixed (removed)

File: src/lbaf/Model/lbsObject.py (hunk 3 above)
Comment (cwschilly) [reply src/lbaf/Model/lbsObject.py]: Fixed (renamed to `entity_property`)

File: src/lbaf/Model/lbsObject.py (hunk 5 above)
Comment (cwschilly) [reply src/lbaf/Model/lbsObject.py]: Fixed (improved docs for qoi getters)

**Review by ppebay — APPROVED** [review]
Thanks for corrections @cwschilly

File: src/lbaf/Model/lbsObject.py (hunk 4 above)
Comment (lifflander) [reply src/lbaf/Model/lbsObject.py:229]: Yes, not necessary, just more explicit

**Review by lifflander — COMMENTED** [review]
Looks good to me!

**Pull request merged**